pkg/backuphelpers/backupvars.go

```diff
@@ -0,0 +1,66 @@
+package backuphelpers
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+)
+
+var (
+	validBackupVars = map[string]struct{}{
+		"schedule":   {},
+		"timezone":   {},
+		"data-dir":   {},
+		"config-dir": {},
+		"backup-dir": {},
+	}
+)
+
+type BackupVar interface {
+	GetBackupVars() *BackupConfig
+}
+
+type BackupConfig struct {
+	enabled  bool
+	backupCR backupv1alpha1.EtcdBackupSpec
+	mux      sync.Mutex
+}
+
+func NewBackupConfig(backupCR backupv1alpha1.EtcdBackupSpec) *BackupConfig {
+	return &BackupConfig{
+		enabled:  false,
+		backupCR: backupCR,
+		mux:      sync.Mutex{},
+	}
+}
+
+func (b *BackupConfig) GetBackupVars() *BackupConfig {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	c := NewBackupConfig(b.backupCR)
+	c.enabled = b.enabled
+	return c
+}
+
+func (b *BackupConfig) ArgString() string {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	var args []string
```

Comment:
why so complicated?

```suggestion
        args := []string{fmt.Sprintf(""--pvc="%s", b.backupCR.PvcName), 
           ...
         }
	return strings.Join(args,"\n")
```

doesn't compile, but I hope you get the gist

Reply:
yeah that's stupid, so we have to add tabs in there to match the formatting. Please add a unit test for that in the targetconfigcontroller.

I would suggest (pseudocode):

```
args := []string{}
if b.enabled {
  args = append(args, "\t--enabled=true\n")
  args = append(args, "\t--pvc=${PVC_NAME}\n")
   ...
} else {
  args = append(args, "\t--enabled=false\n")
}
```

maybe we can add the tab with the join, or some other function to prepend it for each entry.

Reply:
ok, but i actually think we dont need the PVC since we use a pre-defined `hostPath`, no ?

Reply:
dude, it's an example :rofl: fill what you need

Reply:
:D :D

Reply:
there is no other way to add the tabs :/

Reply:
> strings.Join(args,"\n\t")  + "\n"

doesn't work? :)

Reply:
not exactly, I had to try various ways to get it right, must have empty strings in the slice beforehand in order to get it right at the end :) 

fixed 👍🏽